main.tf

```
@@ -39,57 +39,86 @@ resource "aws_lambda_function" "lambda_asg" {
   timeout          = 30
 }
 
-
-resource "aws_cloudwatch_event_rule" "asg_downscale_scheduler_event" {
-  count               = var.asg_scheduler != null ? 1 : 0
-  name                = "asg-scheduler-downscale-event-${random_id.this[0].id}"
-  description         = "ASG downscale scheduler for ${var.asg_scheduler.asg_name}"
-  schedule_expression = var.asg_scheduler.downscale_cron_expression
-}
-
-resource "aws_lambda_permission" "allow_asg_downscale" {
-  count         = var.asg_scheduler != null ? 1 : 0
-  action        = "lambda:InvokeFunction"
-  function_name = aws_lambda_function.lambda_asg[0].function_name
-  principal     = "events.amazonaws.com"
-  source_arn    = aws_cloudwatch_event_rule.asg_downscale_scheduler_event[0].arn
-}
-
-resource "aws_cloudwatch_event_target" "asg_downscale_scheduler_target" {
-  count     = var.asg_scheduler != null ? 1 : 0
-  rule      = aws_cloudwatch_event_rule.asg_downscale_scheduler_event[0].name
-  target_id = aws_lambda_function.lambda_asg[0].function_name
-  arn       = aws_lambda_function.lambda_asg[0].arn
-  input = jsonencode({
-    asg_name         = var.asg_scheduler.asg_name
-    desired_capacity = var.asg_scheduler.downscale_desired_capacity
+resource "aws_iam_role" "asg_scheduler" {
+  count = var.asg_scheduler != null ? 1 : 0
+  name  = "${random_id.this[0].id}-asg-scheduler-role"
```

```suggestion
  name  = "asg-scheduler-${random_id.this[0].id}"
```
personal preference so all asg-scheduler-* are grouped with sorting alphabetically
(and we know it's a "role"